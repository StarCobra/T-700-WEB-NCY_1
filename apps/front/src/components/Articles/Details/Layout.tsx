import React from "react";
import "../../../style/article.scss";
import ArticleContent from "./ArticleContent";
import { Box, Button } from "@mui/material";
import { useArticle } from "../../../Context/ArticleProvider";
import { Link } from "react-router-dom";

export default function Layout() {
  const { currentArticle } = useArticle();
  console.log("currentArticle", currentArticle);
  return (
    <Box className="fullArticleDisplay">
      <Box className="articleHeader">
        <Box className="articleThumbnail">
          <img src={currentArticle?.enclosure?.[0]?.$?.url} alt="" />
        </Box>

        <Box className="articleTitleContainer">
          <Box className="title">
            <h3>{currentArticle?.title}</h3>
          </Box>

          <Box className="dateAndSource">
            <h4>{currentArticle?.["dc:creator"]}</h4>
          </Box>
        </Box>
      </Box>

      <ArticleContent article={currentArticle} />

      <Link className="backButton" to={"/articles"}>
        <Button>Back</Button>
      </Link>
    </Box>
  );
}
