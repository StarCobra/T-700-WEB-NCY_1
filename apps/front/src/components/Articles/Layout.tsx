import React from "react";
import "../../style/article.scss";
import OneArticle from "./OneArticle";
import { Box } from "@mui/material";

export default function Layout() {
  return (
    <Box className="articleMasterContainer">
      <OneArticle />

      <OneArticle />

      <OneArticle />

      <OneArticle />
    </Box>
  );
}
