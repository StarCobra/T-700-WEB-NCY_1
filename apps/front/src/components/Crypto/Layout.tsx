import React from "react";
import CryptoArray from "../Crypto/Array";
import CryptoChart from "../Crypto/Chart";
import Select from "../Select";
import { Box, Button } from "@mui/material";
import { Link } from "react-router-dom";

export default function Layout() {
  const [valueCrypto, setValueCrypto] = React.useState("bitcoin");

  // TODO : remplacer par valeur de l'API
  const options = [
    { value: "bitcoin", label: "BTC" },
    { value: "ethereum", label: "ETH" },
    { value: "solana", label: "SOL" },
  ];

  // TODO : remplacer par valeur de l'API
  const resource = [
    {
      id: "BTC",
      current_price: 50000,
      percent_price_day: 0.5,
      highest_price_day: 55000,
      lowest_price_day: 45000,
    },
    {
      id: "ETH",
      current_price: 5000,
      percent_price_day: 0.5,
      highest_price_day: 5500,
      lowest_price_day: 4500,
    },
    {
      id: "SOL",
      current_price: 500,
      percent_price_day: 0.5,
      highest_price_day: 550,
      lowest_price_day: 450,
    },
    {
      id: "ADA",
      current_price: 5,
      percent_price_day: 0.5,
      highest_price_day: 5.5,
      lowest_price_day: 4.5,
    },
    {
      id: "DOT",
      current_price: 50,
      percent_price_day: 0.5,
      highest_price_day: 55,
      lowest_price_day: 45,
    },
    {
      id: "LUNA",
      current_price: 50,
      percent_price_day: 0.5,
      highest_price_day: 55,
      lowest_price_day: 45,
    },
    {
      id: "LINK",
      current_price: 50,
      percent_price_day: 0.5,
      highest_price_day: 55,
      lowest_price_day: 45,
    },
  ];

  // TODO : remplacer par valeur de l'API
  const test = [
    {
      id: "BTC",
      change_price_day: 6629.81,
      current_price: 6650.5,
      highest_price_day: 6623.04,
      lowest_price_day: 6633.33,
    },
    {
      id: "BTC",
      change_price_day: 6610,
      current_price: 6600,
      highest_price_day: 6630,
      lowest_price_day: 6550,
    },
    {
      id: "BTC",
      change_price_day: 6629.81,
      current_price: 6650.5,
      highest_price_day: 6623.04,
      lowest_price_day: 6633.33,
    },
    {
      id: "BTC",
      change_price_day: 6610,
      current_price: 6600,
      highest_price_day: 6630,
      lowest_price_day: 6550,
    },
    {
      id: "BTC",
      change_price_day: 6629.81,
      current_price: 6650.5,
      highest_price_day: 6623.04,
      lowest_price_day: 6633.33,
    },
    {
      id: "BTC",
      change_price_day: 6610,
      current_price: 6600,
      highest_price_day: 6630,
      lowest_price_day: 6550,
    },
    {
      id: "BTC",
      change_price_day: 6629.81,
      current_price: 6650.5,
      highest_price_day: 6623.04,
      lowest_price_day: 6633.33,
    },
    {
      id: "BTC",
      change_price_day: 6610,
      current_price: 6600,
      highest_price_day: 6630,
      lowest_price_day: 6550,
    },
  ];

  return (
    <Box className="ArrayDisplay">
      <Box>
        <Select
          label="Select a crypto(s):"
          options={options}
          handleChange={(e: any) => setValueCrypto(e.target.value)}
        />

        <Button>
          <Link to={"/preferences"}>Update preferences</Link>
        </Button>
      </Box>

      <CryptoChart
        resource={test}
        title={
          valueCrypto !== ""
            ? `CandleStick of ${valueCrypto}`
            : "Select a crypto"
        }
      />

      <Box>
        <h3>Table of your preferences crypto</h3>
        <CryptoArray resource={resource} />
      </Box>
    </Box>
  );
}
