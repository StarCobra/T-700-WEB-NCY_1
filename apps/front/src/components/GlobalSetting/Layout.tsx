import * as React from "react";
import "../../style/admin.scss";
import { Box } from "@mui/material";
import SettingsIcon from "@mui/icons-material/Settings";
import ManagementCrypto from "./ManagementCrypto";
import useAuth from "../../Context/UserProvider";
import ManagementKeyword from "./ManagementKeyword";

export default function Layout() {
  const { userToken } = useAuth();

  return (
    <Box className="ArrayDisplay">
      <Box className="titleContainer">
        <Box className="title">
          <h1>Global Settings</h1>
        </Box>
        <Box className="icon">
          <SettingsIcon />
        </Box>
      </Box>

      <Box>
        <ManagementCrypto userToken={userToken} />

        <ManagementKeyword userToken={userToken} />
      </Box>

      <Box className="commentContainer">
        <h3>
          {"Please "}
          <a href="mailto:developpers@cryptofanatix.com?subject=Request&body=Hello, I have a request for you, ">
            contact a developper
          </a>
          {" if you have an idea or a problem."}
        </h3>
      </Box>
    </Box>
  );
}
