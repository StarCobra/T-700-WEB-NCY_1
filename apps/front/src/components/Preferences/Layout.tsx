import React from "react";
import { Box } from "@mui/material";
import CryptoArray from "../Crypto/Array";
import Select from "../Select";
import "../../style/preferencies.scss";
import "../../style/cryptoDisplay.scss";
import api from "../../services/api";
import Loader from "../Loader";

export default function Layout() {
  const [favoriteCrypto, setFavoriteCrypto] = React.useState([]);

  const [responseAllCrypto, setResponseAllCrypto] = React.useState([] as any);
  const [loadingAllCrypto, setLoadingAllCrypto] = React.useState(true);

  // Requete pour récupérer toutes les cryptos
  React.useEffect(() => {
    const fetchData = async () => {
      try {
        setLoadingAllCrypto(true);
        const data = await api.getAllCryptos();
        setResponseAllCrypto(data);
      } catch (error) {
        console.error("Error fetching cryptos:", error);
      } finally {
        setLoadingAllCrypto(false);
      }
    };

    fetchData();
  }, []);

  return (
    <Box className="preferencesLayout">
      <Box className="headerContainer">
        <h3>Preferences page</h3>
        <small>
          This page is used to manage your preferences. You can choose your
          favorite crypto.
        </small>
        <small>
          The chosen crypto will be displayed on the home page and you&apos;ll
          receive in priority the new articles about it.
        </small>
      </Box>
      <Box className="arrayContainer">
        {loadingAllCrypto ? (
          <Loader />
        ) : (
          <>
            <h3>Table of cryptos available</h3>
            <CryptoArray resource={responseAllCrypto.data} />
          </>
        )}
      </Box>
      <Box className="addPrefContainer">
        <Box className="selectContainer">
          <h3>Add crypto to your favorite cypto</h3>
          <Select
            name="Select"
            label={
              loadingAllCrypto
                ? "Loading crypto..."
                : "Favorite crypto(s) chosen:"
            }
            type="multiple"
            options={responseAllCrypto.data}
            forCrypto={true}
            handleChange={(e: any) => setFavoriteCrypto(e.target.value)}
          />
        </Box>
        <Box className="cryptoDisplayContainer">
          <h3>
            You have chosen {favoriteCrypto.length} crypto(s) as favorite crypto
            :
          </h3>
          {favoriteCrypto.length > 0 && (
            <ul>
              {favoriteCrypto.map((crypto: any) => (
                <li key={crypto}>{crypto}</li>
              ))}
            </ul>
          )}
        </Box>
      </Box>
    </Box>
  );
}
