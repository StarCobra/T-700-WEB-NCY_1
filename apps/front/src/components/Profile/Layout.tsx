import React from "react";
import "../../style/profile.scss";
import ProfileInformation from "./Information";
import { Box } from "@mui/material";
import exampleImage from "../../assets/profile/profileExample.png";

export default function Layout() {
  return (
    <Box className="centeringDiv">
      <Box className="profileLayout">
        <Box className="profileThumbnailContainer">
          <img src={exampleImage} alt="" />
        </Box>

        <ProfileInformation />
      </Box>
    </Box>
  );
}
