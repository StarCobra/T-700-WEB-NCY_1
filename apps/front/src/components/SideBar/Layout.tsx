import React from "react";
import User from "./User";
import Menu from "./Menu";
import "../../style/Default/default.scss";
import "../../style/sidebar.scss";
import { Box } from "@mui/material";
import Logo from "../../assets/sideBar/logo.png";
import { Link } from "react-router-dom";

export default function Layout() {
  const user = null;
  const isLogged: boolean = user !== null;
  let userComponent;
  if (isLogged) {
    userComponent = <User user={user} />;
  }
  return (
    <Box className="sideBarContainer">
      <Link to="/" className="logoLocation">
        <img src={Logo} alt="" />
      </Link>

      {userComponent}

      <Menu user={user} />
    </Box>
  );
}
