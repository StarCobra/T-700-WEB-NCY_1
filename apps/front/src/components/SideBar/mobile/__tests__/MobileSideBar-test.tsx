import React from "react";
import { act, create } from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Layout from "../MobileSideBar";

describe("Components > SideBar > mobile > MobileSideBar", () => {
  const tests = [
    {
      description: "generic",
      props: {},
    },
  ];
  tests.forEach((test) => {
    it(`should render the component ${test.description}`, async () => {
      let root;

      await act(async () => {
        root = create(
          <MemoryRouter>
            <Layout {...test.props} />
          </MemoryRouter>,
        );
      });

      expect(root).toMatchSnapshot();
    });
  });
});
