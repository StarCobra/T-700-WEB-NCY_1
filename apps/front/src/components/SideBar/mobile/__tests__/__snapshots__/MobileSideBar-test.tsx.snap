// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > SideBar > mobile > MobileSideBar should render the component generic 1`] = `
<div
  className="mobileSideBarLayout"
>
  <div
    className="mobileLogo"
  >
    <img
      src="default"
    />
  </div>
  <div
    className="mobileSandwichLogo"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium Icons css-i4bv87-MuiSvgIcon-root"
      data-testid="MenuIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
  <div
    className="mobileSignOut"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium Icons css-i4bv87-MuiSvgIcon-root"
      data-testid="LogoutIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
      />
    </svg>
  </div>
</div>
`;
